[aoti] refactor aoti_torch__scaled_mm and skip aoti fp8 test for some cases

henrylhtsang opened the pull request:
Continuing https://github.com/pytorch/pytorch/pull/128683 and https://github.com/pytorch/pytorch/pull/130582.

The api of _scaled_mm has changed. For example, there is only one return now. So change the aoti api as well. 

Also, tested the fp8 tests offline. The test_fp8_abi_compatible_cpu_with_stack_allocation_and_minimal_arrayref_interface would fail with `error: use of undeclared identifier 'float8_e4m3fn'` and `error: use of undeclared identifier 'half'`, so skipping them for now.

The reason this wasn't known earlier is probably because the CI doesn't use H100.

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @ColinPeppler @amjames @desertfire @chauhang

chenyang78 reviewed (changes requested):
On a second thought, we may actually want to add a new API with "_v2" suffix, since changing the arguments breaks the backward compatibility. 

cc @desertfire

henrylhtsang commented:
@chenyang78 I added the v2 functions. Should I modify the implementation of v1 to use that of v2 as well? 

btw does aot inductor know to use v2 automatically?

henrylhtsang commented:
Verified with internal testing that the tests are passing

henrylhtsang committed:
[aoti] refactor _scaled_mm and skip fp8 test for some cases (#130868)

Summary: Pull Request resolved: https://github.com/pytorch/pytorch/pull/130868

Differential Revision: D59833074

chenyang78 reviewed (approved):
LGTM. Thanks!

henrylhtsang commented:
@pytorchbot label "topic: not user facing"

henrylhtsang commented:
@pytorchbot merge

henrylhtsang commented:
@pytorchbot merge
